tools/x-repo-types/README.md

```diff
@@ -0,0 +1,67 @@
+# Cross Repo Type Comparisons
+
+Given two types **X** and **Y** from separate repositories, compare the types and generate a report of any differences to the serialized shape of the types. In particular it ignores different embedding of structs, different field names if `codec` tags are used, and different types if they map to the same primitives.
+This tool is designed to be used in CI systems to alert us if a change is made to one repo without a corresponding change to another. For example the `Genesis` type in `go-algorand` and `go-algorand-sdk`. See the [Makefile](./Makefile) for additional examples.
+
+## Build the `xrt` binary
+
+```sh
+make build-xrt
+```
+
+## Example run
+
+```sh
+./xrt --x-repo "github.com/algorand/go-algorand" \
+    --x-package "github.com/algorand/go-algorand/ledger/ledgercore" \
+    --x-type "StateDelta" \
+    --y-repo "github.com/algorand/go-algorand-sdk/v2@develop" \
```

Comment:
typo? I don't see `y-repo` arg in rootCmd

Reply:
Need to look into this.

Reply:
Seems to be a weird artifact. The [current commit](https://github.com/algorand/go-algorand/blob/73318220e88daff8085a175d9b0171688d6aa80b/tools/x-repo-types/Makefile#L10-L15) no longer has `y-repo`:

Reply:
then it is a typo. it's in the README.

Reply:
Dohhh, thx!

Reply:
fix in [5c1777f](https://github.com/algorand/go-algorand/pull/5304/commits/5c1777f4e639d7d1ed12c8cd66ab65cbb1489d29)